pkg/collector/prometheus_process_collector.go

```diff
@@ -0,0 +1,285 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package collector
+
+import (
+	"strconv"
+	"sync"
+
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/sustainable-computing-io/kepler/pkg/bpfassets/attacher"
+	collector_metric "github.com/sustainable-computing-io/kepler/pkg/collector/metric"
+	"github.com/sustainable-computing-io/kepler/pkg/config"
+)
+
+type processDesc struct {
```

Comment:
My only other concern is that like other containers, system process container looks like it also records Cgroup stats. However this is not being exported in the general prometheus collector for new metrics or in the prometheus_process_collector.go as referenced. In a future PR, can we consider including CGroup related metrics instead of only the Hardware Counters.

Reply:
It is a debatable use case. On environment no cgroups are configured, the cgroup metrics may not be available for processes.

Reply:
I see. In that case soft irq metrics and counter metrics are the way to go. The reason I ask is because we are exploring setting up package power models for counter metrics, cgroup features, and irq metrics.